src/Servers/Kestrel/Core/src/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -531,5 +531,22 @@ private static bool IsHex(char ch)
                 // Check if less than 6 representing chars 'a' - 'f'
                 || (uint)((ch | 32) - 'a') < 6u;
         }
+
+        public static AltSvcHeader? GetEndpointAltSvc(System.Net.IPEndPoint endpoint, HttpProtocols protocols)
+        {
+            var hasHttp1OrHttp2 = protocols.HasFlag(HttpProtocols.Http1) || protocols.HasFlag(HttpProtocols.Http2);
+            var hasHttp3 = protocols.HasFlag(HttpProtocols.Http3);
+
+            if (hasHttp1OrHttp2 && hasHttp3)
+            {
+                var text = $"h3=\":{endpoint.Port}\"; ma=84600";
+                var bytes = Encoding.ASCII.GetBytes($"\r\nAlt-Svc: {text}");
```

Comment:
This is called per connection, right?
I'd expect string concat to be more efficient.
```suggestion
                var bytes = Encoding.ASCII.GetBytes("\r\nAlt-Svc: h3=\":" + endpoint.Port.ToString(InvariantCulture) + "\"; ma=84600");
```